bin/build.js

```diff
@@ -202,6 +203,44 @@ export async function build() {
 const copyStaticAssets = () =>
   step('📋 Copying static assets', () => fs.copy('static', 'dist/static'));
 
+const buildTemplatesJson = () =>
+  step('Generating templates json', async () => {
+    const root = 'static/templates';
+    const result = {};
+
+    for (const name of await fs.readdir(root)) {
+      const fileUrl = file => `${root}/${name}/${file}`;
```

Comment:
Lets add a leading slash to this url for safety.

Reply:
Actually, scratch that. Let's simplify:

1. All `html` files are `index.html`. Let's prescind `contentUrl` and generate on the client.
2. All `preview` files are `preview.*`. Let's keep only the extension and rewrite `TemplatePreview` to render a video when `mp4`.

```js
    for (const name of await fs.readdir(root)) {
      let previewExt;
      let files;

      const shouldIncludeFile = file => {
        if (file.startsWith('_preview.')) {
          previewExt = file.split('.').pop();
          return false;
        }
        return !/(^\.|\.(html|txt|md)$)/.test(file);
      };

      try {
        files = (await fs.readdir(`${root}/${name}`)).filter(shouldIncludeFile);
      } catch (_) {
        continue;
      }

      result[name] = {files, previewExt};
    }
```

And change `TemplatePreview` accordingly:

```js

const templateFileUrl = (templateName, filename) =>
  `/static/templates/${templateName}/${filename}`;

const TemplateSelector = ({name, previewExt}) => html`
  <div
    class="${n('template')}"
    @click=${dispatchSelectTemplate}
    data-name=${name}
  >
    ${TemplatePreview(name, previewExt)}
  </div>
`;

function TemplatePreview(name, ext) {
  const url = templateFileUrl(name, `_preview.${ext}`);
  if (ext == 'mp4' || ext == 'webm') {
    return html`
      <video autoplay loop muted src=${url}></video>
    `;
  }
  return html`
    <img src=${url} />
  `;
}
```

And use the same `templateFileUrl` function for getting the `index.html`.